d2/src/main/java/com/linkedin/d2/balancer/clients/TrackerClientImpl.java

```diff
@@ -307,4 +309,17 @@ else if (throwable instanceof StreamException)
     }
     return false;
   }
+
+  private CallCompletion startCall(Request request)
+  {
+    String retryHeader = request.getHeader(HttpConstants.HEADER_NUMBER_OF_RETRY_ATTEMPTS);
+    if (retryHeader != null && Integer.parseInt(retryHeader) > 0)
+    {
+      return _callTracker.startCall(true);
+    }
```

Comment:
I'm a little bit confused, what's the purpose of tracking # of retries in TrackerClient here? Each trackerClient corresponds to one server host, do we need to track retry count on server host basis?

Reply:
This change is reverted